src/buf/fixed/pool.rs

```diff
@@ -323,15 +348,8 @@ impl FixedBuffers for Inner {
         unsafe { slice::from_raw_parts(self.raw_bufs.as_ptr(), self.states.len()) }
     }
 
-    fn check_in(&mut self, index: u16, init_len: usize) {
-        let cap = self.iovecs()[index as usize].iov_len;
-        let state = &mut self.states[index as usize];
-        debug_assert!(
-            matches!(state, BufState::CheckedOut),
-            "the buffer must be checked out"
-        );
-        let next = self.free_buf_head_by_cap.insert(cap, index);
-        *state = BufState::Free { init_len, next };
+    unsafe fn check_in(&mut self, index: u16, init_len: usize) {
```

Comment:
No SAFETY comment?

Reply:
The implementation is delegated to an all-safe method, no additional concerns here.
The caller's safety requirements are documented on the trait.